rpcserver.go

```diff
@@ -1885,6 +2075,71 @@ func marshalRoute(route *routing.Route) *lnrpc.Route {
 	return resp
 }
 
+func unmarshalRoute(route *lnrpc.Route, graph *channeldb.ChannelGraph) (*routing.Route, error) {
+	req := &routing.Route{
+		TotalTimeLock: route.TotalTimeLock,
+		TotalFees:     btcutil.Amount(route.TotalFees),
+		TotalAmount:   btcutil.Amount(route.TotalAmt),
+		Hops:          make([]*routing.Hop, len(route.Hops)),
+	}
+
+	sourceNode, err := graph.SourceNode()
+	if err != nil {
+		return nil, fmt.Errorf("unable to featch source node from the graph "+
+			"while unmarshaling routes: %v", err)
+	}
+
+	var nextSourceNode *channeldb.LightningNode
+
+	for i, hop := range route.Hops {
+		_, c1, c2, err := graph.FetchChannelEdgesByID(hop.ChanId)
+		if err != nil {
+			return nil, fmt.Errorf("unable to featch channel edges by channel ID "+
+				"for hop(%d): %v", i, err)
+		}
+
+		var channelEdgePolicy *channeldb.ChannelEdgePolicy
+
+		if nextSourceNode == nil {
```

Comment:
Oh, the devil made me do it :)
Yes, this case is real unnecessarily complicated. 
I have made changes according to all your latest comments.
Thanks!